### PR TITLE
Update ewsv2 exc2019

### PR DESCRIPTION
**Pull request by ostolero**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3086)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml [hunk 1]
```diff
@@ -122,8 +122,8 @@ configuration:
   section: Connect
   advanced: true
 description: Exchange Web Services and Office 365 (mail)
-display: EWS v2
-name: EWS v2
+display: EWS_v2_2019_version
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml:125]: we shouldnt change display name as the integration is compatible with 2013 and 2016 too

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -1031,6 +1054,19 @@ def email_ec(item):
...
 
+def parse_object_as_dict_with_serialized_items(object):
+    raw_dict = {}
+    if object is not None:
+        for field in object.FIELDS:
+            try:
+                v = getattr(object, field.name, None)
+                json.dumps(v)
+                raw_dict[field.name] = v
+            except (TypeError, OverflowError):
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:1065]: lets add a debug log that an exception was caught

**Review by JudahSchwartz — COMMENTED** [review]
Great work on this 1! 🚀

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml (hunk 1 above)
Comment (ostolero) [reply Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.yml:125]: oops just for testing, reverting

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 3]
```diff
@@ -1606,7 +1634,7 @@ def delete_items(item_ids, delete_type, target_mailbox=None):     # pragma: no c
 
 
 def prepare_args(d):      # pragma: no cover
-    d = dict((k.replace("-", "_"), v) for k, v in d.items())
+    d = dict((k.replace("-", "_"), v) for k, v in list(d.items()))
     if 'is_public' in d:
         if exchangelib.__version__ != "1.12.0":  # Docker BC
```
[the remark this answers is not in the document]
Comment (ostolero) [reply Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:1639]: Removing and assuming __version__ is not 1.12.0

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 4]
```diff
@@ -21,37 +21,52 @@
...
-                                 EXCHANGE_2016)
+                                 EXCHANGE_2016, EXCHANGE_2019)
 from future import utils as future_utils
 from requests.exceptions import ConnectionError
+from exchangelib.version import VERSIONS as EXC_VERSIONS
+
+
+# Exchange2 2019 patch - server dosen't connect with 2019 but with other versions creating an error mismatch (see CIAC-3086),
+# overriding this function to prevent error printed when running in debug-mode.
+def our_fullname(self):
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:38]: please open a bug on exchangelib for this and link it here

File: Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py [hunk 5]
```diff
@@ -21,37 +21,52 @@
...
                                  EXCHANGE_2010_SP2, EXCHANGE_2013,
-                                 EXCHANGE_2016)
+                                 EXCHANGE_2016, EXCHANGE_2019)
 from future import utils as future_utils
 from requests.exceptions import ConnectionError
+from exchangelib.version import VERSIONS as EXC_VERSIONS
+
+
+# Exchange2 2019 patch - server dosen't connect with 2019 but with other versions creating an error mismatch (see CIAC-3086),
+# overriding this function to prevent error printed when running in debug-mode.
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py:37]: lets explain what exactly we changed

File: Packs/MicrosoftExchangeOnPremise/ReleaseNotes/2_0_0.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+
+#### Integrations
+
+##### EWS v2
+- Updated the Docker image to: *demisto/py-ews:5.0.2.63879*.
+Updated integration to support Exchange 2019
```
Comment (JudahSchwartz) [inline Packs/MicrosoftExchangeOnPremise/ReleaseNotes/2_0_0.md:6]: lets also add that it was switched to python3

**Review by JudahSchwartz — APPROVED** [review]
awesome! בשעה טובה. Just minor comments here

**Pull request merged**